### DISCUSSION TITLE
Can i use this app with root mode ?

### DISCUSSION THREAD

**Discussion by xDoge26** [lihenggui/blocker#140] [Q&A] [answered]
And what is Controller type and online source rule ?

**Comment by lihenggui** [comment, marked as answer]
You can use this app with root mode. 

Blocker supports three types of component controllers: PackageManager, IFW, and Shizuku. You can seamlessly switch between these controllers, all of which are designed to control the enable/disable status of an application component. The default controller is IFW (Intent Firewall), and I recommend using it.

The online source rule is the general rule for searching a component. If an application integrates a 3rd party SDK, some components will be integrated into the app. The general rule is used to search for common components. There are two servers for storing the general rule: GitHub and GitLab. Because most users are in China and can't access GitHub, I have set up GitLab as a backup for Chinese users.
